src/services/users.js

```diff
@@ -0,0 +1,165 @@
+        return newUser ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user already exists');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async getUsers (req, res) {
+    const { _id, name, email, status, role, skip, take } =
+      (typeof req.query !== 'undefined' && req.query) || {};
+    const query = {};
+    if (_id != null) query._id = _id;
+    if (name != null) query.name = name;
+    if (email != null) query.email = email;
+    if (status != null) query.status = status;
+    if (role != null) query.role = role;
+    if (skip != null) query.skip = skip;
+    if (take != null) query.take = take;
+
+    try {
+      const users = await User.find(query)
+        .select('-password')
+        .skip(+query.skip || 0)
+        .limit(+query.take || 50)
+        .exec();
+
+      return users ? res.json(users) : [];
+    } catch (err) {
+      return res.boom.badRequest();
+    }
+  }
+
+  async updateUser (req, res) {
+    try {
+      const updated = await User.findByIdAndUpdate(req.params.id, req.body, {
+        new: true
+      }).exec();
+      return updated ? res.json(true) : res.boom.notFound();
+    } catch (err) {
+      return res.boom.notFound();
+    }
+  }
+
+  async getRoleAdmin (req, res) {
+    const roleAdmin = await UserRoles.findOne({ name: 'Admin' });
+    return roleAdmin;
+  }
+
+  async createAdmin (req, res) {
+    try {
+      const roleAdmin = await this.getRoleAdmin();
+      const statusActive = await this.getStatusActive();
+      const newAdmin = await new User(req.body);
+      newAdmin.role = roleAdmin._id;
+      newAdmin.status = statusActive._id;
+      const salt = await bcrypt.genSalt(10);
+      newAdmin.password = await bcrypt.hash(newAdmin.password, salt);
+      const userExist = await User.findOne({ email: req.body.email });
+      if (!userExist) {
+        await newAdmin.save();
+        return newAdmin ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user already exists');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async getStatusBlocked (req, res) {
+    const statusBlocked = await UserStatuses.findOne({ name: 'Blocked' });
+    return statusBlocked;
+  }
+
+  async blockUser (req, res) {
+    try {
+      const statusBlocked = await this.getStatusBlocked();
+      const user = await User.findByIdAndUpdate(
+        req.params.id,
+        { status: statusBlocked._id },
+        {
+          new: true
+        }
+      ).exec();
+      return user ? res.json(true) : res.boom.notFound();
+    } catch (err) {
+      res.boom.notFound();
+    }
+  }
+
+  async getStatusDeleted (req, res) {
+    const statusDeleted = await UserStatuses.findOne({ name: 'Deleted' });
+    return statusDeleted;
+  }
+
+  async deleteUser (req, res) {
+    try {
+      const statusDeleted = await this.getStatusDeleted();
+      const user = await User.findByIdAndUpdate(
```

Comment:
What if user is already blocked or deleted?

Reply:
fixed